### PR TITLE
fix(plugin): test plugin functionality

### PR DESCRIPTION
**Pull request by scefali**
This PR fixes the issue when pressing `Test Plugin` would result in success even though there was actually an error. I've also taken the opportunity to improve the error messaging and standardization of error handling of plugins so the error you see in `Test Plugin` is meaningful.

Before:

Screen Shot 2020-02-11 at 1 56 00 PM

After:

Screen Shot 2020-02-11 at 1 52 32 PM

### CODE REVIEW

File: src/sentry_plugins/slack/plugin.py [hunk 1]
```diff
@@ -138,7 +138,7 @@ def get_tag_list(self, name, project):
             return None
         return set(tag.strip().lower() for tag in option.split(","))
 
-    def notify(self, notification):
+    def notify(self, notification, raise_exception=False):
```
Comment (MeredithAnya) [inline src/sentry_plugins/slack/plugin.py]: How come for some of these you use `**kwargs` and here you explicitly have `raise_exception=False`?

Comment (scefali) [reply src/sentry_plugins/slack/plugin.py]: @MeredithAnya I used `**kwargs` when I didn't actually use the parameter. I can switch it if you think it makes more sense though.

Comment (scefali) [reply src/sentry_plugins/slack/plugin.py]: @MeredithAnya I believe `**kwargs` is used then you have arbitrary extra fields. See what we have on the base class for notify: https://github.com/getsentry/sentry/blob/master/src/sentry/plugins/base/notifier.py#L9

Comment (MeredithAnya) [reply src/sentry_plugins/slack/plugin.py]: cool that wfm

**Review by MeredithAnya — APPROVED** [review]

**Pull request merged**